### PR TITLE
Filter out non balancer pools by checking the vault

### PR DESCRIPTION
**Pull request by yvesfracari**

### CODE REVIEW

File: scripts/listen_to_events/main.py [hunk 1]
```diff
@@ -183,7 +187,7 @@ async def main():
         tasks = [setup_and_run_chain(chain) for chain in NOTIFICATION_CHAIN_MAP.keys()]
         await asyncio.gather(*tasks, start_discord_bot())
     except Exception as e:
-        sentry_sdk.capture_exception(e)
+        # sentry_sdk.capture_exception(e)
```
Comment (ribeirojose) [inline scripts/listen_to_events/main.py]: 👀

File: scripts/listen_to_events/main.py [hunk 2]
```diff
@@ -48,14 +48,18 @@
...
     data = {
         "chain": chain,
         "event": event,
     }
-
+    
     strategy = STRATEGY_MAP.get(parse_event_name(data["event"]), DefaultEventStrategy)()
+    
+    if not await strategy.check_if_pool_has_right_vault(chain, event):
+        logger.info(f"Pool {event['address']} does not have the right vault")
```
Comment (ribeirojose) [inline scripts/listen_to_events/main.py]:
```suggestion
        logger.info(f"Pool {event['address']} is not registered in Balancer's official Vault")
```

File: scripts/listen_to_events/main.py [hunk 3]
```diff
@@ -48,14 +48,18 @@
...
     """Handle the event, parse it, and send a notification."""
     data = {
         "chain": chain,
         "event": event,
     }
-
+    
     strategy = STRATEGY_MAP.get(parse_event_name(data["event"]), DefaultEventStrategy)()
+    
+    if not await strategy.check_if_pool_has_right_vault(chain, event):
```
Comment (ribeirojose) [inline scripts/listen_to_events/main.py]:
```suggestion
    if not await strategy.is_from_balancer_dao(chain, event):
```

File: scripts/listen_to_events/strategies.py [hunk 4]
```diff
@@ -130,6 +130,21 @@ async def format_data(self, _chain, data):
...
+    async def get_pool_address(self):
+        raise NotImplementedError("Subclasses should implement this method")
+
+    async def check_if_pool_has_right_vault(self, chain, event):
+        web3 = Web3Provider.get_instance(chain, {}, NOTIFICATION_CHAIN_MAP)
+        pool_address = web3.to_checksum_address(await self.get_pool_address(event))
+        try:
+            pool = web3.eth.contract(address=pool_address, abi=get_mock_pool_abi())
+            pool_vault = await pool.functions.getVault().call()
+            return pool_vault == "0xBA12222222228d8Ba445958a75a0704d566BF2C8"
```
Comment (ribeirojose) [inline scripts/listen_to_events/strategies.py]: `#would` normalize/lowercase both addresses 🤷  I never know what case these libs are going to throw as us

File: scripts/listen_to_events/strategies.py [hunk 5]
```diff
@@ -130,6 +130,21 @@ async def format_data(self, _chain, data):
...
         raise NotImplementedError("Subclasses should implement this method")
+    
+    async def get_pool_address(self):
+        raise NotImplementedError("Subclasses should implement this method")
+
+    async def check_if_pool_has_right_vault(self, chain, event):
+        web3 = Web3Provider.get_instance(chain, {}, NOTIFICATION_CHAIN_MAP)
+        pool_address = web3.to_checksum_address(await self.get_pool_address(event))
+        try:
+            pool = web3.eth.contract(address=pool_address, abi=get_mock_pool_abi())
```
Comment (ribeirojose) [inline scripts/listen_to_events/strategies.py:141]: if this fails we'll also throw. Should this be the case?

File: scripts/listen_to_events/strategies.py [hunk 6]
```diff
@@ -203,6 +224,9 @@ class AmpUpdateStoppedStrategy(EventStrategy):
     async def format_topics(self, chain, event):
         return {k: v for k, v in parse_event_topics(event).items()}
 
+    async def get_pool_address(self, event):
+        return event["address"]
```
Comment (ribeirojose) [inline scripts/listen_to_events/strategies.py:228]: don't use deceitful function names
```suggestion
    async def get_event_address(self, event):
        return event["address"]
```

File: scripts/listen_to_events/strategies.py [hunk 7]
```diff
@@ -275,11 +299,19 @@ async def format_data(self, chain, event):
 
     def discord_channels(self):
         return os.getenv("BALANCER_NEW_POOL_DISCORD_CHANNEL_IDS", "").split(",")
+    
+    async def get_pool_address(self, event):
+        data = parse_event_topics(event)
+        return "0x" + data["poolAddress"][-40:]
```
Comment (ribeirojose) [inline scripts/listen_to_events/strategies.py:305]: 👀 why the cutoff at -40?

**Review by ribeirojose — APPROVED** [review]

File: scripts/listen_to_events/strategies.py (hunk 6 above)
Comment (yvesfracari) [reply scripts/listen_to_events/strategies.py:228]: but not always the pool and the event will be the same address. For example, the factory emits an event of a new pool being created.

File: scripts/listen_to_events/strategies.py (hunk 5 above)
Comment (yvesfracari) [reply scripts/listen_to_events/strategies.py:141]: The main reason for failure, in my view, is a wrong abi. IMO in this case we shouldn't post it.

File: scripts/listen_to_events/strategies.py (hunk 7 above)
Comment (yvesfracari) [reply scripts/listen_to_events/strategies.py:305]: Im not sure, I got this from other function on this same `Strategy`

File: scripts/listen_to_events/strategies.py (hunk 6 above)
Comment (ribeirojose) [reply scripts/listen_to_events/strategies.py:228]: right, that's a bit my point 👀 this function doesn't know what it's extracting from the event.

**Pull request merged**